tests/test_api_events_traversal.py

```
@@ -0,0 +1,37 @@
+from swarm.api.routes.events import generate_run_events
+
+@pytest.mark.anyio
+async def test_generate_run_events_traversal(tmp_path):
+    """Test that generate_run_events raises ValueError for path traversal."""
+    runs_root = tmp_path / "runs"
+    runs_root.mkdir()
+
+    # These should raise ValueError immediately
+    with pytest.raises(ValueError, match="run_id"):
+        gen = generate_run_events("../etc", runs_root)
+        await gen.__anext__()
```

Ruff (0.14.14)

[warning] `pytest.raises()` block should contain a single simple statement

(PT012)